statsmodels/tools/testing.py

```diff
@@ -16,8 +16,10 @@ def strip_rc(version):
 
 
 def is_pandas_min_version(min_version):
-    '''check whether pandas is at least min_version '''
-    return StrictVersion((pandas.__version__[:6])) >= min_version
+    '''check whether pandas is at least min_version
+    '''
+    from pandas import __version__ as pversion
+    return StrictVersion(strip_rc(pversion)) >= min_version
```

Comment:
my point about the attitude was your way of checking is simply wrong. The following works just fine with and w/o dev versions.

```
In [6]: LooseVersion('0.17.0+126.gacf0eed') < '0.17.0'
Out[6]: False

In [7]: LooseVersion('0.17.0') < '0.17.0'
Out[7]: False

In [8]: LooseVersion('0.17.0') < '0.18.0'
Out[8]: True

In [9]: LooseVersion('0.17.0+126.gacf0eed') < '0.18.0'
Out[9]: True
```